File: examples/currency/convert/node/convert10UsdToGbp.js
import * as currency from "@m3o/services/currency";

// Convert returns the currency conversion rate between two pairs e.g USD/GBP
async function Convert10usdToGbp() {
  let currencyService = new currency.CurrencyService(
    process.env.MICRO_API_TOKEN
  );
  let rsp = await currencyService.convert({
    amount: 10,
    from: "USD",
    to: "GBP",
  });
  console.log(rsp);
}

await Convert10usdToGbp();

File: examples/currency/convert/go/convert10UsdToGbp.go
package example

import (
	"fmt"
	"os"

	"github.com/micro/services/clients/go/currency"
)

// Convert returns the currency conversion rate between two pairs e.g USD/GBP
func Convert10usdToGbp() {
	currencyService := currency.NewCurrencyService(os.Getenv("MICRO_API_TOKEN"))
	rsp, err := currencyService.Convert(&currency.ConvertRequest{
		Amount: 10,
		From:   "USD",
		To:     "GBP",
	})
	fmt.Println(rsp, err)
}
